tests/monster_test.fbs

```diff
@@ -141,6 +141,14 @@ table Monster {
   // enum value.
   long_enum_non_enum_default:LongEnum (id: 52);
   long_enum_normal_default:LongEnum = LongOne (id: 53);
+  // Test that default values nan and +/-inf work.
+  nan_default:float = nan (id: 54);
```

Comment:
We probably should. It's honestly such a pain to add fields to this test file (multiple languages hard-code the expected fields in the test) that I'm almost inclined not to....

Reply:
True. I think the direction we want to go in w.r.t. testing is to write new schema files for each feature/test. I think that'd be cleaner for this change but obviously it'll take a little boilerplate to set up all the languages and test that they compile.

This pattern is helpful for advanced features that aren't implemented in every language at the same time, but I see monster_test as the "basic features" that all languages should support so I can see why you'd test this here.

Its up to you which you think is easier

Reply:
I just added a single double field to exercise the logic (and because I was lazy about getting to this, apparently I also got to fix nim's handling of this issue....).